solc/CommandLineInterface.cpp

```diff
@@ -692,7 +694,12 @@ void CommandLineInterface::compile()
 {
 	solAssert(CompilerInputModes.count(m_options.input.mode) == 1);
 
-	m_compiler = make_unique<CompilerStack>(m_fileReader.reader());
+	auto modelChecker = m_options.modelChecker.settings;
+	auto universalCallback{UniversalCallback(m_fileReader, SMTSolverCommand("eld"))};
```

Comment:
This leaves us with two copies of the file reader (`m_fileReader` and the one inside the universal callback) which can easily go out of sync. This is IMO just asking for trouble. The command-line interface keeps referring to `m_fileReader`, while the callback pulls files from its own copy.

Reply:
I know, but if we put it inside the branch it causes exceptions.

Reply:
I think it should be fine. `m_fileReader` is only used as read at line 769: `m_compiler->setSources(m_fileReader.sourceUnits());`, and after that the compiler is already called. I don't see how the objects can go out of sync.

Reply:
Ultimately, I can also change this callback to go directly into the `ModelCheckerSettings` only as `SMTSolverCommand` instead of the compiler-wide `ReadCallback`, since it doesn't really use the C callback but is only a wrapper.